components/inputs/input-date-range.js

```diff
@@ -95,18 +103,21 @@ class InputDateRange extends RtlMixin(LocalizeCoreElement(LitElement)) {
 		if (!this.label) {
 			console.warn('d2l-input-date-range component requires label text');
 		}
+
+		this._endLabel = this.endLabel ? this.endLabel : this.localize('components.input-date-range.endDate');
+		this._startLabel = this.startLabel ? this.startLabel : this.localize('components.input-date-range.startDate');
 	}
 
 	render() {
-		const startLabel = this.startLabel ? this.startLabel : this.localize('components.input-date-range.startDate');
-		const endLabel = this.endLabel ? this.endLabel : this.localize('components.input-date-range.endDate');
 		return html`
+			<d2l-validation-custom @d2l-validation-custom-validate=${this._validateRange}></d2l-validation-custom>
 			<d2l-input-fieldset label="${ifDefined(this.label)}" ?label-hidden="${this.labelHidden}">
 				<d2l-input-date
 					@change="${this._handleChange}"
 					class="d2l-input-date-range-start"
 					?disabled="${this.disabled}"
-					label="${startLabel}"
+					?invalid="${this.validationError ? true : false}"
```

Comment:
This will likely need to change so that its only passed in if the `validationError` is present. If it isn't present then we don't want to provide the error at all so that the component itself can set `invalid`.

Reply:
Once my PR gets in to add `invalid` and `forceInvalid` you'll be able to change these to:
```
.forceInvalid=${this.invalid}
```